src/log_file.rs

```diff
@@ -27,11 +30,7 @@ impl LogFile {
         // realistically it seems unlikely to be hit often...
         let basename = clean_filename(scenario_name);
         for i in 0..1000 {
-            let t = if i == 0 {
-                format!("{}.log", basename)
-            } else {
-                format!("{}_{:03}.log", basename, i)
-            };
+            let t = Self::log_file_name(&basename, i);
             let path = log_dir.join(t);
             match OpenOptions::new()
```

Comment:
Do you get a clear error message here like `ENAMETOOLONG`? If so I wonder if we could truncate the names only after seeing that error, so that we still have the full and more-useful names on other filesystems?